### ISSUE TITLE
Do you have any solution to enable RTSP in TP-link NC 250 model?

### ISSUE THREAD

**Issue by sovietmah** [reald/nc220#4]
Hi expert, do you have any solution to enable RTSP in TP-link NC 250 model? This is the latest model from TP-link

**Comment by teknotronix** [comment]
Hi,

I own an NC250 and found that the links provided (https://github.com/reald/nc220) work for the NC250 as well. However, I found a bug in the firmware where you have to change the Frame Rate setting in the camera's web page, hit save and then the RTSP stream starts working. You can change it back to your desired Frame Rate again but It needs to change and be saved once to get the RTSP stream to work. This needs to be done each time you power up or reboot the camera.

I also found that 30fps does not work at all on the RTSP stream. I use VLC to view the stream.

**Comment by ignisf** [comment]
Ahoy,

Here are my observations, @sovietmah and @teknotronix:

The NC 250 actually has `rtsp://admin:admin@[device-ip]:554/h264_hd.sdp` (thank you binwalk) that can be used when you switch on the device.

Whenever you change the video settings, this stream disappears and moves to `rtsp://admin:admin@[device-ip]:554/h264_vga.sdp`.

I am amazed by TP-Link's engineering ;).

**Comment by sovietmah** [comment]
Cool, sorry, just to double confirm, so you means the original firmware
already support RTSP?

**Comment by ignisf** [comment]
Yes, albeit buggy.

**Comment by sovietmah** [comment]
Thx a lot, i will buy one and try it out this weekend.

**Comment by sovietmah** [comment]
i tried both:
rtsp://admin:admin@192.168.0.106:554/h264_hd.sdp

rtsp://admin:admin@192.168.0.106:554/h264_vga.sdp

using VLC, is not working for me, I already setup the camera and my IP
address is 192.168.0.106, and i still using the default username & password.

**Comment by sovietmah** [comment]
But i can paping the ip camera address 80, 554, 8080 port is working OK.

**Comment by ignisf** [comment]
Works for me with all of these:

```
vlc rtsp://admin:admin@<IP>:554/h264_hd.sdp
mpv rtsp://admin:admin@<IP>:554/h264_hd.sdp
ffplay rtsp://admin:admin@<IP>:554/h264_hd.sdp
```

Are you getting an error message? Maybe try ffplay and see what it says.

**Comment by alexjaya** [comment]
Hi,

The RSTP feed works in VLC for me.
However it doesn't work on Genius Vision.
Anyone has any ideas what I am missing or has managed to get it to work in Genius Vision?

**Comment by morfikov** [comment]
I have NC250 and I've been playing with this camera for a while. It looks like settings in the camera admin panel maters in the case of `h264_vga.sdp` or `h264_hd.sdp`. I was able to get stream using:

`$ cvlc rtsp://admin:admin@192.168.1.159:554/h264_vga.sdp`

But only when the frame rate is set to 10 FPS. The other frame rates don't work with the above command, and I couldn't get `h264_hd.sdp` working at all.

The next thing is the following streams:

```
$ mpv http://admin:YWRtaW4\=@192.168.1.159:8080/stream/video/h264?resolution=HD
$ mpv http://admin:YWRtaW4\=@192.168.1.159:8080/stream/audio/mpegaacblock
```
I got this sniffing the network when I was connecting using the tpcamera smartphone app. The video stream command utilizes only 5-120 KiB/s, whereas the following one around 2500 KiB/s:

`$ mpv --no-cache http://admin:YWRtaW4\=@192.168.1.159:8080/stream/video/mjpeg`

The image is pretty much the same, but using `h264?resolution=HD` I get 30-40s delay, but it works fine. My smartphone works without delay. I don't know how to fix it, besides `mjpeg` works also almost without a delay.

**Comment by sovietmah** [comment]
OK
Now I tried with:
rtsp://admin:admin@[Ip address]:554/h264_hd.sdp using VLC, and it is working.
but unfortunately I still failed to setup RTSP in synology surveillance station. But mpeg succeeded.
https://ahkhai.com/2016/07/31/setup-tp-link-nc-250-on-synology-surveillance-station/

**Comment by ProfDrYoMan** [comment]
Hello, I just got two questions about the N250 too.

Are
http://[ipofcamera]:8080/stream/video/mjpeg
and especially
wget --user=admin --password=YWRtaW4= http://[ipofcamera]:8080/stream/snapshot.jpg
working as for the N220?

In what resolutions are stream and snapshot?

**Comment by icyman61** [comment]
I have several of these cameras and would like to get them working. Has anyone got this to work with Blue Irus?

**Comment by igorfili** [comment]
Yes, I have Blue Iris and one Tplink NC250. Under Network IP Camera configuration select rtsp:// , then fill in with IPADDR/h264_hd.sdp  or IPADDR/h264_vga.sdp. Fill in username and password fields and that should be all.

**Comment by nachopro** [comment]
Hi to all!! I have 5 NC220 cameras. The Onvif Settings by default comes with mjpeg (without sound, obviously).

Under Windows using ONVIF Device Manager (https://sourceforge.net/projects/onvifdm/) you can change the video streaming to h264.

Holly Charm!

**Comment by LucasArruda** [comment]
Thanks. That really helped me with the NC250.
